python/tests/test_tables.py

```diff
@@ -1061,9 +1061,17 @@ def test_simple_example(self):
         t = tskit.MutationTable()
```

Comment:
Looks good, thanks @petrelharp. It'd be nice to add a quick test to check this in isolation, since it's subtle enough to miss in the first place. Something like ``test_missing_time_equal_to_self`` and just using the example you had in the issue.

Reply:
done.